ThingIFSDK/ThingIFSDK/Gateway/GatewayAPI.swift

```diff
@@ -26,7 +26,8 @@ open class GatewayAPI: NSObject, NSCoding {
         return self.gatewayAddress.absoluteString
     }
 
-    private var accessToken: String?
+    /** Access token of this gate way */
+    open private(set) var accessToken: String?
```

Comment:
> We adopted to forgive to override properties and methods in principle.

Yes. We decided to mark as open if the property access level public or open in principle.
Since it is hard to judge needs of override by API clients.

public will be used for a part which breaks the our expectation in design.

> overriding accessToken does not make sense on this Class.

Might be true. But client could override methods and do some logging when the accessToken is nil. And it doesn't breaks our expectation.
It's a silly example but still valid.